ghcide/src/Development/IDE/Plugin/HLS.hs

```diff
@@ -212,8 +218,8 @@ executeCommandHandlers recorder ecs = requestHandler SMethod_WorkspaceExecuteCom
           Nothing -> logAndReturnError recorder p (InR ErrorCodes_InvalidRequest) (commandDoesntExist com p xs)
           Just (PluginCommand _ _ f) -> case A.fromJSON arg of
             A.Error err -> logAndReturnError recorder p (InR ErrorCodes_InvalidParams) (failedToParseArgs com p err arg)
-            A.Success a ->
-                f ide a `catchAny` -- See Note [Exception handling in plugins]
+            A.Success a -> do
+              (first (toResponseError . (p,)) <$> runExceptT (f ide a)) `catchAny` -- See Note [Exception handling in plugins]
```

Comment:
this is pretty hard to read. I'd probably try to make it a bit more "point-ful", maybe use the do block more.